services/metric/manager.go

```diff
@@ -0,0 +1,76 @@
+/*
+Package metric implements an abstraction for safely managing metrics in concurrent environments.
+*/
+package metric
+
+import "fmt"
+
+const (
+	DEFAULT_REGISTRY        string = "default"
+	PENDING_EVENTS_REGISTRY string = "pending_events"
+)
+
+var instance Manager
+
+func init() {
+	reset()
+}
+
+func reset() {
+	instance = &manager{
+		registries: map[string]Registry{},
+	}
+	instance.(*manager).createRegistry(DEFAULT_REGISTRY)
+	instance.(*manager).createRegistry(PENDING_EVENTS_REGISTRY)
+}
+
+// GetManager gets the manager singleton
+func GetManager() Manager {
+	return instance
+}
+
+func Reset() {
+	reset()
+}
+
+// Manager is the entry-point for retrieving metric registries
+type Manager interface {
+	// GetRegistry gets a registry by its key
+	GetRegistry(key string) Registry
+}
+
+type manager struct {
+	registries map[string]Registry
+}
+
+func (r *manager) createRegistry(key string) {
+	r.registries[key] = NewRegistry()
+}
+
+func (r *manager) GetRegistry(key string) Registry {
+	return r.registries[key]
+}
+
+func GetPendingEventsMeasurement(tablePrefix string, workspace string, destType string) Gauge {
+	return GetManager().GetRegistry(PENDING_EVENTS_REGISTRY).MustGetGauge(&pendingEventsMeasurement{tablePrefix, workspace, destType})
```

Comment:
a more general doubt: would using pointers as a key in the map yield expected results..?
`&pendingEventsMeasurement{tablePrefix, workspace, destType}`

Reply:
the Registry accepts an interface as an argument, thus the pointer. It is true that we would be in trouble if using pointers as map keys, thus we use the value returned from the `GetKey` method as a key, which is the actual struct. 
I admit that this is not the simplest/cleanest solution right now, we can further discuss to find a better design

Reply:
oh right. I misread the return in `GetKey`.
This looks alright